dspace-api/src/main/java/org/dspace/content/RelationshipServiceImpl.java

```diff
@@ -346,6 +357,119 @@ public void delete(Context context, Relationship relationship, boolean copyToLef
+        }
+        List<RelationshipType> relationshipTypes = new LinkedList<>();
+        EntityType leftType = relationship.getRelationshipType().getLeftType();
+        String entityTypeStringFromMetadata = relationshipMetadataService.getEntityTypeStringFromMetadata(item);
+        EntityType actualEntityType = entityTypeService.findByEntityType(context, entityTypeStringFromMetadata);
+        boolean isLeft = false;
+        if (StringUtils.equalsIgnoreCase(leftType.getLabel(),
+                                         entityTypeStringFromMetadata)) {
+            relationshipTypes = relationshipTypeService.findByEntityType(context, actualEntityType, false);
+        } else {
+            isLeft = true;
+            relationshipTypes = relationshipTypeService.findByEntityType(context, actualEntityType, true);
+        }
+        for (RelationshipType relationshipType : relationshipTypes) {
+            if (virtualMetadataPopulator.getMap().containsKey(relationshipType.getRightwardType())) {
+                List<Relationship> list = findByItemAndRelationshipType(context, item, relationshipType, isLeft);
+                for (Relationship foundRelationship : list) {
+                    if (isLeft) {
+                        itemsToUpdate.add(foundRelationship.getRightItem());
+                        return getRelatedItemsForRightItem(context, foundRelationship.getRightItem(), foundRelationship,
+                                                           itemsToUpdate, max);
+                    } else {
+                        itemsToUpdate.add(foundRelationship.getLeftItem());
+                        return getRelatedItemsForLeftItem(context, foundRelationship.getLeftItem(), foundRelationship,
+                                                          itemsToUpdate, max);
+                    }
+                }
+            }
+        }
+        return itemsToUpdate;
+    }
+
+    private List<Item> getRelatedItemsForLeftItem(Context context, Item item, Relationship relationship,
+                                                  List<Item> itemsToUpdate, int max)
+        throws SQLException {
+        if (itemsToUpdate.size() >= max) {
+            return itemsToUpdate;
+        }
+        List<RelationshipType> relationshipTypes = new LinkedList<>();
+        EntityType rightType = relationship.getRelationshipType().getRightType();
+        String entityTypeStringFromMetadata = relationshipMetadataService.getEntityTypeStringFromMetadata(item);
+        EntityType actualEntityType = entityTypeService.findByEntityType(context, entityTypeStringFromMetadata);
+        boolean isLeft = false;
+        if (StringUtils.equalsIgnoreCase(rightType.getLabel(),
+                                         relationshipMetadataService.getEntityTypeStringFromMetadata(item))) {
+            isLeft = true;
+            relationshipTypes = relationshipTypeService.findByEntityType(context, actualEntityType, true);
+        } else {
```

Comment:
Same here...the logic is still confusing.  We already know this `item` is the "LeftItem", so when would this `else` ever be triggered?  Why again do we need the `isLeft` variable, if we know it should always be `true`?